python_modules/dagster/dagster/_core/execution/context/invocation.py

```diff
@@ -55,17 +56,58 @@
 from dagster._utils.forked_pdb import ForkedPdb
 from dagster._utils.merger import merge_dicts
 
-from .compute import OpExecutionContext
+from .compute import AssetExecutionContext, OpExecutionContext
 from .system import StepExecutionContext, TypeCheckContext
 
 
 def _property_msg(prop_name: str, method_name: str) -> str:
     return (
-        f"The {prop_name} {method_name} is not set on the context when a solid is directly invoked."
+        f"The {prop_name} {method_name} is not set on the context when an asset or op is directly"
+        " invoked."
     )
 
 
-class UnboundOpExecutionContext(OpExecutionContext):
+class BoundExecutionContext(OpExecutionContext):
```

Comment:
Having this interface be a child of `OpExecutionContext` feels shady, but it's the only way i was able to find to make the type checker happy. 

When AssetExecutionContext eventually splits from OpExecutionContext there will be a good amount of work to do to make the path through `op_invocation.py` happy with the two types